### PR TITLE
Migrate to structured logging 3

### PR DESCRIPTION
**Pull request by sayantani11**
Part 3 of #120

### CODE REVIEW

File: pkg/trimaran/targetloadpacking/targetloadpacking.go [hunk 1]
```diff
@@ -216,7 +216,7 @@ func (pl *TargetLoadPacking) Score(ctx context.Context, cycleState *framework.Cy
 	nodeCPUCapMillis := float64(nodeInfo.Node().Status.Capacity.Cpu().MilliValue())
 	nodeCPUUtilMillis := (nodeCPUUtilPercent / 100) * nodeCPUCapMillis
 
-	klog.V(6).Infof("node %v CPU Utilization (millicores): %v, Capacity: %v", nodeName, nodeCPUUtilMillis, nodeCPUCapMillis)
+	klog.V(6).Infof("node %v CPU Utilization (millicores):%v , Capacity:%v ", nodeName, nodeCPUUtilMillis, nodeCPUCapMillis)
```
[the remark this answers is not in the document]
Comment (sayantani11) [reply pkg/trimaran/targetloadpacking/targetloadpacking.go:219]: Working on the suggestions

File: pkg/trimaran/loadvariationriskbalancing/analysis.go [hunk 2]
```diff
@@ -73,7 +73,8 @@ func (rs *resourceStats) computeScore(margin float64, sensitivity float64) float
 
 	// evaluate overall risk factor
 	risk := (mu + sigma) / 2
-	klog.V(6).Infof("mu=%f; sigma=%f; margin=%f; sensitivity=%f; risk=%f", mu, sigma, margin, sensitivity, risk)
+	klog.V(6).InfoS("mu=", "mu", mu, " sigma=", "s", sigma, " margin=", "margin", margin,
```
Comment (Huang-Wei) [inline pkg/trimaran/loadvariationriskbalancing/analysis.go:76]: This doesn't look right. Structure logging starts with a general message:

```suggestion
	klog.V(6).InfoS("Evaluating risk factor", "mu", mu, "sigma", sigma, "margin", margin, "sensitivity", sensitivity, "risk", risk)
```

File: pkg/trimaran/loadvariationriskbalancing/analysis.go [hunk 3]
```diff
@@ -83,7 +84,7 @@ func createResourceStats(metrics []watcher.Metric, node *v1.Node, podRequest *fr
 	// get resource usage statistics
 	nodeUtil, nodeStd, metricFound := getResourceData(metrics, watcherType)
 	if !metricFound {
-		klog.V(6).Infof("resource %s usage statistics for node %s: no valid data", watcherType, node.GetName())
+		klog.V(6).InfoS("Rsource usage statistics for node : no valid data", " node", klog.KObj(node))
```
Comment (Huang-Wei) [inline pkg/trimaran/loadvariationriskbalancing/analysis.go:87]: remove unneeded spaces, and fix typo:

```suggestion
		klog.V(6).InfoS("Resource usage statistics for node: no valid data", "node", klog.KObj(node))
```

File: pkg/trimaran/loadvariationriskbalancing/analysis.go [hunk 4]
```diff
@@ -105,8 +106,8 @@ func createResourceStats(metrics []watcher.Metric, node *v1.Node, podRequest *fr
 	rs.usedAvg = nodeUtil * rs.capacity / 100
 	rs.usedStdev = nodeStd * rs.capacity / 100
 
-	klog.V(6).Infof("resource %s usage statistics for node %s: capacity=%f; req=%f; usedAvg=%f; usedStdev=%f",
-		watcherType, node.GetName(), rs.capacity, rs.req, rs.usedAvg, rs.usedStdev)
+	klog.V(6).InfoS("resource usage statistics for node:", "node", klog.KObj(node), " capacity", rs.capacity,
```
Comment (Huang-Wei) [inline pkg/trimaran/loadvariationriskbalancing/analysis.go:109]: Capitalize the first word, and remove the ending `:`, as well as empty spaces. (applies elsewhere)

File: pkg/trimaran/loadvariationriskbalancing/collector.go [hunk 5]
```diff
@@ -58,11 +58,10 @@ func newCollector(obj runtime.Object) (*Collector, error) {
 	// get the plugin arguments
 	args := getArgs(obj)
 
-	klog.V(4).Infof("Using LoadVariationRiskBalancingArgs: MetricProvider.Type=%q, MetricProvider.Address=%q,"+
-		" SafeVarianceMargin=%f, SafeVarianceSensitivity=%f, WatcherAddress=%q",
-		args.MetricProvider.Type, args.MetricProvider.Address, args.SafeVarianceMargin,
-		args.SafeVarianceSensitivity, args.WatcherAddress)
+	klog.V(4).InfoS("Using LoadVariationRiskBalancingArgs:", "type", args.MetricProvider.Type, "address", args.MetricProvider.Address,
```
Comment (Huang-Wei) [inline pkg/trimaran/loadvariationriskbalancing/collector.go:61]: Ditto. remove the ending `:`.

File: pkg/trimaran/loadvariationriskbalancing/loadvariationriskbalancing.go [hunk 6]
```diff
@@ -111,14 +111,14 @@ func (pl *LoadVariationRiskBalancing) Score(ctx context.Context, cycleState *fra
 	if cpuOK {
 		cpuScore = cpuStats.computeScore(pl.collector.args.SafeVarianceMargin, pl.collector.args.SafeVarianceSensitivity)
 	}
-	klog.V(6).Infof("pod:%s; node:%s; CPUScore=%f", pod.GetName(), nodeName, cpuScore)
+	klog.V(6).InfoS("pod:Name; node:nodeName; CPUScore=", "Name", klog.KObj(pod), "nodeName", nodeName, "cpuScore", cpuScore)
```
Comment (Huang-Wei) [inline pkg/trimaran/loadvariationriskbalancing/loadvariationriskbalancing.go:114]: Ditto. The first field should be a generic msg:

```suggestion
	klog.V(6).InfoS("CPUScore for pod", "pod", klog.KObj(pod), "nodeName", nodeName, "cpuScore", cpuScore)
```

BTW: I think you also missed migrating L94. Could you do a thorough check?

File: pkg/trimaran/loadvariationriskbalancing/loadvariationriskbalancing.go [hunk 7]
```diff
@@ -111,14 +111,14 @@ func (pl *LoadVariationRiskBalancing) Score(ctx context.Context, cycleState *fra
...
 	// calculate Memory score
 	var memoryScore float64 = 0
 	memoryStats, memoryOK := createResourceStats(metrics, node, podRequest, v1.ResourceMemory, watcher.Memory)
 	if memoryOK {
 		memoryScore = memoryStats.computeScore(pl.collector.args.SafeVarianceMargin, pl.collector.args.SafeVarianceSensitivity)
 	}
-	klog.V(6).Infof("pod:%s; node:%s; MemoryScore=%f", pod.GetName(), nodeName, memoryScore)
+	klog.V(6).InfoS("pod", klog.KObj(pod), " node",  nodeName, " MemoryScore", memoryScore)
```
Comment (Huang-Wei) [inline pkg/trimaran/loadvariationriskbalancing/loadvariationriskbalancing.go:121]:
```suggestion
	klog.V(6).InfoS("MemoryScore for pod", "pod", klog.KObj(pod), "nodeName",  nodeName, "memoryScore", memoryScore)
```

File: pkg/trimaran/loadvariationriskbalancing/loadvariationriskbalancing.go [hunk 8]
```diff
@@ -127,7 +127,7 @@ func (pl *LoadVariationRiskBalancing) Score(ctx context.Context, cycleState *fra
 		totalScore = math.Max(memoryScore, cpuScore)
 	}
 	score = int64(math.Round(totalScore))
-	klog.V(6).Infof("pod:%s; node:%s; TotalScore=%d", pod.GetName(), nodeName, score)
+	klog.V(6).InfoS("pod", klog.KObj(pod), " nodeName", nodeName, " TotalScore", score)
```
Comment (Huang-Wei) [inline pkg/trimaran/loadvariationriskbalancing/loadvariationriskbalancing.go:130]: Ditto. Please ensure:

- it starts with a general message
- remove extra empty spaces in the key fields
- lowercase the key name

File: pkg/trimaran/targetloadpacking/targetloadpacking.go [hunk 9]
```diff
@@ -193,7 +193,7 @@ func (pl *TargetLoadPacking) Score(ctx context.Context, cycleState *framework.Cy
 	for _, container := range pod.Spec.Containers {
 		curPodCPUUsage += PredictUtilisation(&container)
 	}
-	klog.V(6).Infof("predicted utilization for pod %v: %v", pod.Name, curPodCPUUsage)
+	klog.V(6).InfoS("Predicted utilization for pod", "pod", klog.KObj(pod), "curPodCPUUsage", curPodCPUUsage)
```
Comment (Huang-Wei) [inline pkg/trimaran/targetloadpacking/targetloadpacking.go:196]:
```suggestion
	klog.V(6).InfoS("Predicted utilization for pod", "pod", klog.KObj(pod), "cpuUsage", curPodCPUUsage)
```

File: pkg/trimaran/targetloadpacking/targetloadpacking.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Huang-Wei) [reply pkg/trimaran/targetloadpacking/targetloadpacking.go:219]: Still outstanding:

1. Capitalize the first word
2. The first parameter is a general msg, and shouldn't come with `%v`

File: pkg/trimaran/targetloadpacking/targetloadpacking.go [hunk 10]
```diff
@@ -246,13 +246,13 @@ func (pl *TargetLoadPacking) Score(ctx context.Context, cycleState *framework.Cy
 			return framework.MinNodeScore, framework.NewStatus(framework.Success, "")
 		}
 		penalisedScore := int64(math.Round(50 * (100 - predictedCPUUsage) / (100 - float64(hostTargetUtilizationPercent))))
-		klog.V(6).Infof("penalised score for host %v: %v", nodeName, penalisedScore)
+		klog.V(6).InfoS("Penalised score for host", "nodeName", nodeName, "penalisedScore", penalisedScore)
```
Comment (Huang-Wei) [inline pkg/trimaran/targetloadpacking/targetloadpacking.go:249]:
```suggestion
		klog.V(6).InfoS("Penalised score for host", "nodeName", nodeName, "score", penalisedScore)
```

**Comment by Huang-Wei** [comment]
Please also resolve the CI failures.

**Comment by Huang-Wei** [comment]
Ping @sayantani11 .

**Comment by sayantani11** [comment]
@Huang-Wei I'll push the changes, just give me sometime. Was busy with another project. Sorry for the delay.

**Comment by sayantani11** [comment]
I tried looking into the error but kind of lost. Running the `make verify-gofmt` gives this:
```sh
Makefile:127: recipe for target 'verify-gofmt' failed
make: *** [verify-gofmt] Error 1
```

**Comment by Huang-Wei** [comment]
@sayantani11 you can run `hack/update-gofmt.sh` to fix the error. https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_scheduler-plugins/253/pull-scheduler-plugins-verify-gofmt-master/1438929177028857856/build-log.txt gives the detailed info.

**Pull request closed without merging**